common/include/camera/IProjection.hpp

```diff
@@ -0,0 +1,67 @@
+#ifndef _NBL_IPROJECTION_HPP_
+#define _NBL_IPROJECTION_HPP_
+
+#include "nbl/builtin/hlsl/cpp_compat/matrix.hlsl"
+
+namespace nbl::hlsl
+{
+
+template<typename T>
+concept ProjectionMatrix = is_any_of_v<T, float64_t4x4, float32_t4x4, float16_t4x4>;
+
+//! Interface class for projection
+template<ProjectionMatrix T>
+class IProjection : virtual public core::IReferenceCounted
+{
+public:
+    using value_t = T;
+
+    IProjection(const value_t& matrix = {}) : m_projectionMatrix(matrix) { updateHandnessState(); }
+
+    inline void setMatrix(const value_t& projectionMatrix)
+    {
+        m_projectionMatrix = projectionMatrix;
+        updateHandnessState();
+    }
+
+    inline const value_t& getMatrix() { return m_projectionMatrix; }
+    inline bool isLeftHanded() { return m_isLeftHanded; }
+
+private:
+    inline void updateHandnessState()
+    {
+        m_isLeftHanded = hlsl::determinant(m_projectionMatrix) < 0.f;
+    }
+
+    value_t m_projectionMatrix;
```

Comment:
this belongs in `LinearProjection`, the basic `IProjection` should be for anything, thin-lens, spherical, octahedral, etc.